client/src/utils/parser.test.js

```diff
@@ -0,0 +1,351 @@
+            ]
+          },
+          {
+            "_attributes": {
+              "id": "Flow_14488my_di",
+              "bpmnElement": "Flow_14488my"
+            },
+            "di:waypoint": [
+              {
+                "_attributes": {
+                  "x": "520",
+                  "y": "330"
+                }
+              },
+              {
+                "_attributes": {
+                  "x": "580",
+                  "y": "330"
+                }
+              }
+            ]
+          },
+          {
+            "_attributes": {
+              "id": "Flow_1ayp4hv_di",
+              "bpmnElement": "Flow_1ayp4hv"
+            },
+            "di:waypoint": [
+              {
+                "_attributes": {
+                  "x": "680",
+                  "y": "330"
+                }
+              },
+              {
+                "_attributes": {
+                  "x": "742",
+                  "y": "330"
+                }
+              }
+            ]
+          }
+        ]
+      }
+    }
+  }
+};
+const BROWSER_ACTIVITY_NAME = 'Browser Open Browser';
+const EXCEL_ACTIVITY_NAME = 'Excel Open Workbook';
+const LIBRARY_EXCEL = 'Library    RPA.Excel.Files';
+const LIBRARY_BROWSER = 'Library    RPA.Browser';
+const SETTING_STRING = '*** Settings ***';
+const TASK_STRING = '*** Tasks ***';
+const OPEN_WORKBOOK_CMD = 'Open Workbook';
+const FIND_EMPTY_ROW_CMD = 'Find Empty Row';
+const OPEN_BROWSER_CMD = 'Open Browser';
+
+describe('Parsing Tests', () => {
+  test('Manually pass in BPMN Json', () => {
+    const parserResultString = parser.parseDiagramJson(JSON_FOR_TESTING);
+
+    expect.assertions(15);
+    expect(parserResultString).toMatch(SETTING_STRING);
+    expect(parserResultString).toMatch(LIBRARY_EXCEL);
+    expect(parserResultString).toMatch(LIBRARY_BROWSER);
+
+    expect(parserResultString).toMatch(EXCEL_ACTIVITY_NAME);
+    expect(parserResultString).toMatch(OPEN_WORKBOOK_CMD);
+    expect(parserResultString).toMatch(FIND_EMPTY_ROW_CMD);
+
+    expect(parserResultString).toMatch(BROWSER_ACTIVITY_NAME);
+    expect(parserResultString).toMatch(OPEN_BROWSER_CMD);
+
+    expect(parserResultString.indexOf(BROWSER_ACTIVITY_NAME)).toBeLessThan(parserResultString.lastIndexOf(OPEN_BROWSER_CMD));
+    expect(parserResultString.indexOf(EXCEL_ACTIVITY_NAME)).toBeLessThan(parserResultString.lastIndexOf(OPEN_WORKBOOK_CMD));
+    expect(parserResultString.indexOf(EXCEL_ACTIVITY_NAME)).toBeLessThan(parserResultString.indexOf(FIND_EMPTY_ROW_CMD));
+
+    expect(parserResultString.indexOf(SETTING_STRING)).toBeLessThan(parserResultString.indexOf(TASK_STRING));
+    expect(parserResultString.indexOf(LIBRARY_BROWSER)).toBeLessThan(parserResultString.indexOf(TASK_STRING));
+    expect(parserResultString.indexOf(LIBRARY_EXCEL)).toBeLessThan(parserResultString.indexOf(LIBRARY_BROWSER));
+    expect(parserResultString.indexOf(LIBRARY_EXCEL)).toBeLessThan(parserResultString.indexOf(TASK_STRING));
```

Comment:
normally I'd criticize that these are too many expect statements for the test. Seeing the upcoming change of the parser I will wave it through. A possible solution for this should be multiple tests (one test for containment and one for the order of the statements) or functions. Does outsourcing to functions work in tests?